### PR TITLE
[WIP] Hk allow unpublished posts

### PR DESCRIPTION
**Pull request by hanneskaeufler**
- [ ] Wait on https://github.com/luckyframework/lucky_record/issues/177

### CODE REVIEW

File: src/models/published_post.cr [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class PublishedPost
+  def initialize(@post : Post)
+  end
+
+  def id
+    @post.id
+  end
+
+  def created_at
```
Comment (hanneskaeufler) [inline src/models/published_post.cr]: There's probably like a macro to do this ...

**Comment by hk-picky** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: No JUnit file was found at ~/reports/output.xml
```
 #  from Dangerfile:19
 #  -------------------------------------------
 #  
 >  junit.parse "~/reports/output.xml"
 #  junit.report
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by hk-picky** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: No JUnit file was found at /root/reports/output.xml
```
 #  from Dangerfile:18
 #  -------------------------------------------
 #  
 >  junit.parse "/root/reports/output.xml"
 #  junit.report
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: src/models/post.cr [hunk 2]
```diff
@@ -2,7 +2,7 @@ class Post < BaseModel
   table :posts do
     column title : String
     column content : String
-    column published_at : Time
+    column published_at : Time?
```
Comment (hanneskaeufler) [inline src/models/post.cr]: I dont' like what this change brings with it ... :(

**Comment by hanneskaeufler** [comment]
Wont do since migrated to nextjs/vercel

**Pull request closed without merging**